### PR TITLE
Use sizeof instead of handcrafted arithmetic

### PR DESCRIPTION
**Pull request by casperisfine**
Also removing the `DISABLE_UNPACKER_STACK_RMEM` flag, as I don't see its value. 

@peterzhu2118

### CODE REVIEW

File: ext/msgpack/unpacker.h [hunk 1]
```diff
@@ -49,7 +49,7 @@ struct msgpack_unpacker_stack_t {
     msgpack_unpacker_stack_t *parent;
 };
 
-#define MSGPACK_UNPACKER_STACK_SIZE (8+4+8+8)  /* assumes size_t <= 64bit, enum <= 32bit, VALUE <= 64bit */
+#define MSGPACK_UNPACKER_STACK_SIZE sizeof(msgpack_unpacker_stack_entry_t)
```
Comment (peterzhu2118) [inline ext/msgpack/unpacker.h:52]: This is a good catch, due to alignment `sizeof(msgpack_unpacker_stack_entry_t)` should be 32 rather than 28.

**Review by peterzhu2118 — APPROVED** [review]

**Comment by casperisfine** [comment]
Ah damnit, I see why it doesn't use sizeof now, it's because it need that value in the preprocessor.

I still think we can likely clean that up somehow.

**Comment by peterzhu2118** [comment]
This looks good.

> so maybe we'd me better to add a static assertion on msgpack_unpacker_stack_entry_t size

tbh I think that this is a good idea if we never want to turn off `UNPACKER_STACK_RMEM`.

**Comment by casperisfine** [comment]
Closing in favor of https://github.com/msgpack/msgpack-ruby/pull/324

**Pull request closed without merging**